lampshade_test.go

```diff
@@ -357,23 +358,23 @@ func TestSessionPool(t *testing.T) {
 	var wg sync.WaitGroup
 	dialNTimes(wg, maxStreamsPerConn)
 	wg.Wait()
-	assert.EqualValues(t, 1, rd.getNumLivePending())
+	assert.EqualValues(t, 1, rd.getNumLivePending(), "Opening up to MaxID should have resulted in 1 session")
 
-	log.Debug("******Setting delay to 2 second")
-	atomic.StoreInt64(&delay, int64(2*time.Second))
+	atomic.StoreInt64(&delay, int64(10*redialSessionInterval))
 	dialNTimes(wg, maxStreamsPerConn)
-	time.Sleep(1 * time.Second)
+	time.Sleep(5 * redialSessionInterval)
 	assert.EqualValues(t, maxLiveConns, rd.getNumLivePending(), "Should dial up to MaxLiveConns sessions when network becomes unusable")
 
-	log.Debug("******Clearing delay")
 	atomic.StoreInt64(&delay, 0)
 	wg.Wait() // Make sure streams can be created after network recovers
 	assert.EqualValues(t, maxLiveConns, rd.getNumLivePending(), "Should keep the live sessions when network recovers")
 
-	time.Sleep(2 * time.Second)
-	dialNTimes(wg, maxStreamsPerConn)
+	time.Sleep(10 * redialSessionInterval)
+	dialNTimes(wg, maxLiveConns-1) // drain all but one of the live sessions
 	wg.Wait()
-	assert.EqualValues(t, 1, rd.getNumLivePending(), "Only one live session should be left after idling for a while")
+	time.Sleep(10 * redialSessionInterval)
```

Comment:
Adding this to make tests more reliable (prevent `getNumLivePending()` below from getting an earlier value)